libs/pika/async_cuda/include/pika/async_cuda/then_with_stream.hpp

```diff
@@ -566,11 +562,7 @@ namespace pika::cuda::experimental::then_with_stream_detail {
     /// This is a helper that calls f with the values sent by sender and a
     /// cuda_stream as the last argument.
     template <typename Sender, typename F>
-    auto then_with_cuda_stream(Sender&& sender, F&& f) -> decltype(
```

Comment:
I've removed them since they were only used to be able choose which `then_with_x` algorithm to dispatch to in DLA-Future. Since that's no longer required, and the SFINAE makes error messages worse for the cases where the dispatch isn't useful I've removed them.